### PR TITLE
Display search result

### PR DESCRIPTION
**Pull request by hamedbahrami479**
Hi Everybody, 

a code review would be very much appreciated. I decided after a discussion with a fellow student, that the search button does not serve any added value, which is why it is erased for now. But I anyhow left the Button.js file in order to build a button for the user to state a new product in case none of the categories matches what he wants to dispose. 

Thanks and best wishes
Hamed

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -4,27 +4,35 @@ import Header from './Components/Header';
...
+            class="Profile Picture"
+            src={ProfilePicture}
+            alt="better be a pic"
+            width="30%"
+            height="30%"
+            justify-content="center"
+            display="flex"
+          />
+        </article>
+        {/* DropDown werden die Daten von productData.js übergeben: App.js -> DropDown.js */}
```
Comment (julialoeschel) [inline client/src/App.js]: please delete unnecessary code here, especially in { }. If your code is super long and hard to understand, you may leave a comment like this. In all other cases, let the code speak for itself :)

File: client/src/App.js [hunk 2]
```diff
@@ -4,27 +4,35 @@ import Header from './Components/Header';
...
+        <DropDown products={products} />
```
Comment (julialoeschel) [inline client/src/App.js:27]: I see you are using `<br>`everywhere. Don´t! 
please style with css.

File: client/src/App.js [hunk 3]
```diff
@@ -4,27 +4,35 @@ import Header from './Components/Header';
...
         <br></br>
-        {product}
-        <br></br>
-        <br></br>
-        <SearchButton />
-        <h2 className="handling">Best Handling would be:</h2>
+        {/* <SearchButton /> */}
```
Comment (julialoeschel) [inline client/src/App.js]: same :)

File: client/src/Components/DropDown.js [hunk 4]
```diff
@@ -11,8 +13,8 @@ function DropDown() {
         name="selectProduct"
         className="custom-select"
         id="category-select"
-        value={product}
-        onChange={e => setProduct(e.target.value)}
+        value={isSelected_DropDown}
+        onChange={e => setIsSelected_DropDown(e.target.value)}
```
Comment (julialoeschel) [inline client/src/Components/DropDown.js:18]: its better practice to write out event, not just e

File: client/src/Components/ResultBox.js [hunk 5]
```diff
@@ -1,8 +1,63 @@
...
+  const [productData, setProductData] = useState([]);
 
+  useEffect(() => {
+    setProductData(products);
+  });
+
+  const product_handling = productData
+    .filter(data => data.productcategory === isSelected_DropDown)
+    .map(data => data.handling);
+  console.log('Output handling', product_handling);
```
Comment (julialoeschel) [inline client/src/Components/ResultBox.js:14]: for the PRs try to clear out the console.logs.

File: client/src/Components/ResultBox.js [hunk 6]
```diff
@@ -1,8 +1,63 @@
...
 
 
-const ResultBox= () => {
-  return <input className="resultBox" type="text" id="result" name="result"></input>;
-};
+function ResultBox({ products, isSelected_DropDown }) {
+  const [productData, setProductData] = useState([]);
 
+  useEffect(() => {
+    setProductData(products);
```
Comment (julialoeschel) [inline client/src/Components/ResultBox.js:8]: this looks like a unnecessary step to me. are you sure you need this?

File: client/src/Components/ResultBox.js [hunk 7]
```diff
@@ -1,8 +1,63 @@
...
+const InputContainer = styled.section`
+  size: 10ch;
+  background-color: rgb(0, 225, 255);
+  border-radius: 2rem;
+  height: 10rem;
+  width: 80%;
+  margin: auto;
+  display: block;
+`;
+// tag image gegen img ausgetauscht <-----------
```
Comment (julialoeschel) [inline client/src/Components/ResultBox.js:71]: :) pls delete

File: client/src/Components/button.css [hunk 8]
```diff
@@ -1,11 +1,11 @@
-.container
+/* .container
```
Comment (julialoeschel) [inline client/src/Components/button.css:1]: dont push code you dont need :)

File: client/src/Components/button.js [hunk 9]
```diff
@@ -2,24 +2,20 @@ import React, { useEffect, useState } from 'react';
...
-
 const SearchButton = () => {
   return (
-    <div className="container">
       <button>Search</button>
-    </div>
   );
 };
+
+// const ButtonStyled = styled.button`
```
Comment (julialoeschel) [inline client/src/Components/button.js:11]: same

**Review by julialoeschel — COMMENTED** [review]
Hey There 🐜 
good job!
please consider my suggestions 🐟

File: client/src/App.js (hunk 2 above)
Comment (hamedbahrami479) [reply client/src/App.js:24]: Done. All <br> are removed and replaced by proper adjustments via css.

File: client/src/App.js (hunk 3 above)
Comment (hamedbahrami479) [reply client/src/App.js]: done.

File: client/src/Components/ResultBox.js (hunk 5 above)
Comment (hamedbahrami479) [reply client/src/Components/ResultBox.js:14]: Done.

File: client/src/Components/button.css (hunk 8 above)
Comment (hamedbahrami479) [reply client/src/Components/button.css:1]: indeed. and not I have delete. :) Sorry 4the mess :(

File: client/src/Components/button.js (hunk 9 above)
Comment (hamedbahrami479) [reply client/src/Components/button.js:8]: these lines are actually kept for the button, which will now follow. If u don't mind I would leave these settings for now and delete them once and for all, when my button is ready for the page routings.

**Comment by hamedbahrami479** [comment]
Very helpful hints. thanks a lot! :)

**Pull request merged**